pymc3/distributions/dist_math.py

```diff
@@ -29,8 +29,16 @@ def bound(logp, *conditions):
 
 
 def alltrue(vals):
-    return tt.all([tt.all(1 * val) for val in vals])
+    """
+    Asserts truth of all elements in vals, across the lowest axis. This maintains
+    element-wise evaluations for multivariate inputs.
+    """
+    try:
+        return tt.all(tt.stack([1*val for val in vals]), axis=0)
+    except (TypeError, IndexError):
+        return tt.all([tt.all(1 * val) for val in vals])
```

Comment:
You can ravel all variables after safe converting them in theano, then concatenate and finally use tt.all